src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -303,6 +303,25 @@ private void addToGroups(List<BibEntry> entries, Collection<GroupTreeNode> group
         }
     }
 
+    public void checkCrossRefKeys(List<BibEntry> entries) {
+        for (BibEntry possibleParentEntry : entries) {
+            for (BibEntry entry : bibDatabaseContext.getDatabase().getEntries()) {
+                Optional<String> crossRef = entry.getField(StandardField.CROSSREF);
+                Optional<String> childYear = entry.getField(StandardField.YEAR);
+                Optional<String> parentKey = possibleParentEntry.getCitationKey();
+                if (crossRef.isPresent() && crossRef.equals(parentKey) && childYear.isEmpty()) {
```

Comment:
This logic with years is wrong.

You see, there are citation key patterns in JabRef, and by default it's something like `[author][year]`, so every generated citation key will consists of the last name of the first author and a year of paper. It's the default, and that is why you added year logic there. But, for example, I may decide not to include a year, maybe I decided to make citation pattern to look like a title of the paper.

Reply:
It is true. In JabRef one can customize the citationkey patterns extensively. See https://docs.jabref.org/setup/citationkeypatterns

Reply:
We have the class CitationKeyGenerator who does this all under the hood e..g generateKey method